AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Subscribe/Support/ModelSortTests.swift

```diff
@@ -0,0 +1,334 @@
+                      QPredGen(name: "name", myBool: false)]
+        models.sortModels(by: QuerySortBy.ascending(QPredGen.keys.myBool).sortDescriptor, modelSchema: QPredGen.schema)
+        XCTAssertEqual(models[0].myBool, nil)
+        XCTAssertEqual(models[1].myBool, false)
+        XCTAssertEqual(models[2].myBool, true)
+        models.sortModels(by: QuerySortBy.descending(QPredGen.keys.myBool).sortDescriptor, modelSchema: QPredGen.schema)
+        XCTAssertEqual(models[0].myBool, true)
+        XCTAssertEqual(models[1].myBool, false)
+        XCTAssertEqual(models[2].myBool, nil)
+    }
+
+    func testSortModelEnum() {
+        var models = [createModel(enum: .bar),
+                      createModel(enum: .foo),
+                      createModel(enum: .bar)]
+        models.sortModels(by: QuerySortBy.ascending(ExampleWithEveryType.keys.enumField).sortDescriptor,
+                          modelSchema: ExampleWithEveryType.schema)
+        XCTAssertEqual(models[0].enumField, .bar)
+        XCTAssertEqual(models[1].enumField, .bar)
+        XCTAssertEqual(models[2].enumField, .foo)
+        models.sortModels(by: QuerySortBy.descending(ExampleWithEveryType.keys.enumField).sortDescriptor,
+                          modelSchema: ExampleWithEveryType.schema)
+        XCTAssertEqual(models[0].enumField, .foo)
+        XCTAssertEqual(models[1].enumField, .bar)
+        XCTAssertEqual(models[2].enumField, .bar)
+    }
+
+    func testSortModelOptionalEnum() {
+        var posts = [createPost(status: .draft),
+                     createPost(status: .private),
+                     createPost(status: .published),
+                     createPost(status: nil)]
+        posts.sortModels(by: QuerySortBy.ascending(Post.keys.status).sortDescriptor, modelSchema: Post.schema)
+        XCTAssertEqual(posts[0].status, nil)
+        XCTAssertEqual(posts[1].status, .draft)
+        XCTAssertEqual(posts[2].status, .private)
+        XCTAssertEqual(posts[3].status, .published)
+        posts.sortModels(by: QuerySortBy.descending(Post.keys.status).sortDescriptor, modelSchema: Post.schema)
+        XCTAssertEqual(posts[0].status, .published)
+        XCTAssertEqual(posts[1].status, .private)
+        XCTAssertEqual(posts[2].status, .draft)
+        XCTAssertEqual(posts[3].status, nil)
+    }
+
+    func testSortByTwoFields() {
+        let dateTime1 = Temporal.DateTime.now()
+        let dateTime2 = Temporal.DateTime.now().add(value: 1, to: .day)
+        let dateTime3 = Temporal.DateTime.now().add(value: 2, to: .day)
+        var posts = [createPost(rating: 1.0, createdAt: dateTime2),
+                     createPost(rating: 1.0, createdAt: dateTime1),
+                     createPost(rating: 2.0, createdAt: dateTime3)]
+
+        posts.sortModels(by: QuerySortBy.ascending(Post.keys.rating).sortDescriptor, modelSchema: Post.schema)
+        XCTAssertEqual(posts[0].rating, 1.0)
+        XCTAssertEqual(posts[0].createdAt, dateTime2)
```

Comment:
nit: a small note that the order of dateTime in the original array would be preserved when rating is used for comparison can be helpful here